src/main/kotlin/nl/connectplay/scoreplay/data/DatabaseFollowGameRepository.kt

```diff
@@ -0,0 +1,96 @@
+package nl.connectplay.scoreplay.data
+
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.datetime.toKotlinLocalDate
+import nl.connectplay.scoreplay.abstraction.data.FollowGameRepository
+import nl.connectplay.scoreplay.models.dto.GameDto
+import nl.connectplay.scoreplay.models.dto.user.UserDto
+
+class DatabaseFollowGameRepository(private val database: Database) : FollowGameRepository {
+    override suspend fun getFollowedGames(userId: Int, offset: Int, limit: Int?): List<GameDto> {
+        return coroutineScope {
+            async {
+                database.connection?.use { connection ->
+                    val sql = """
+                        SELECT games.game_id AS game_id,
+                               games.name,
+                               games.description,
+                               games.publisher,
+                               games.minimum_player_count,
+                               games.maximum_player_count,
+                               games.duration,
+                               games.minimum_age,
+                               games.release_date
+                        FROM game_followers
+                        JOIN games ON game_followers.game_id = games.game_id
+                        WHERE game_followers.user_id = ?
+                        LIMIT ? OFFSET ?
+                    """.trimIndent()
+
+
+                    val stmt = connection.prepareStatement(sql)
+                    // with the apply block you dont have to define stmt multiple times
+                    stmt.apply {
+                        setInt(1, userId)
+                        setInt(2, limit ?: 10)
+                        setInt(3, offset)
+                    }
+
+                    // returned rows result
+                    val resultSet = stmt?.executeQuery()
+
+                    val games = mutableListOf<GameDto>()
+
+                    while (resultSet?.next() == true) {
+                        val gameId = resultSet.getInt("game_id") // because of the ALIAS its game_id not games_id
+                        val name = resultSet.getString("name")
+                        val description = resultSet.getString("description")
+                        val publisher = resultSet.getString("publisher")
+                        val minPlayers = resultSet.getInt("minimum_player_count")
+                        val maxPlayers = resultSet.getInt("maximum_player_count")
+                        val duration = resultSet.getInt("duration")
+                        val minAge = resultSet.getInt("minimum_age")
```

Comment:
getInt() returnt 0 voor waardes die op de database null zijn (in java zijn primitieve types nooit nullable, maar in kotlin wel)  
kijk dit af in de DatabaseGameRepository